### ISSUE TITLE
Request for adding property created from template to OMO

### ISSUE THREAD

**Issue by wdduncan** [information-artifact-ontology/ontology-metadata#117]
### Label

created from template

### Definition of the property

An annotation property that provides the name of the template or an IRI of the template used to create a term.

### What is the range of the property in question?

xsd:string

### Examples of use

A contrived example from UBERON:
'lower central secondary incisor tooth' `created from template` lower_tooth_pattern.yaml.

### Motivation to add

Ontology terms can come from a number of sources: imports, ROBOT templates, DOSDP templates, and perhaps other custom build templates. In #60, `dc:source` and `created_by` are discussed as possibilities. It would great to get some resolution about this.

cc @matentzn @cmungall 

### ORCID, ROR or Wikidata identifier of the contributor

https://orcid.org/0000-0001-9625-1899

### OMO Checklist

- [X] I believe the property is generally useful beyond my specific ontology needs.
- [X] There is no other property in OMO that covers the same use case.

**Comment by bpeters42** [comment]
I agree that such a property can be useful; but should the template be identified by a URI / purl? Just the name of the template seems to be a bit weak - I would not know where to find the one from this example.

**Comment by wdduncan** [comment]
@bpeters42 In the definition I do mention using an IRI.  

> An annotation property that provides the name of the template or an **IRI** of the template used to create a term.

IRI would be better. But, for purposes of maintaining an ontology, the name may suffice. If something like this was adopted, it may even be best practice to only have the annotation in the edit file, and then remove those annotations when the release is built.

**Comment by cmungall** [comment]
agreed - all our templates have URIs

@matentzn we had previously discussed using dc:conformsTo for this. In fact we do this for classes auto-inferred to belong to a template (or to instantiate a metaclass, which is the better way to think of it). I think we should either

- have a single property and use reification to annotate provenance
- OR have a hierarchy:
     - dc:conformsTo
         - assertedToConformTo
         - generatedFrom
         - detectedAsConformingTo

**Comment by wdduncan** [comment]
> this is always generated by the system

@cmungall Are you referring to UBERON? The lower teeth are generated from a dosdp template, but I am seeing any annotated information about the template, just an obscure database_cross_reference to `OBOL:automatic`.

You may not want to strip them out of the release for UBERON. That is your call. But, I am not so sure I would want to have them in the OHD release. The information is mainly intended for editors (in my case). But, I am willing to think about this more. In any case, how the annotations are maintained is a different subject.

**Comment by matentzn** [comment]
I regret a bit that we are using http://www.geneontology.org/formats/oboInOwl#source in dosdp tools at the moment. 

In Mondo we use dc:conformsTo and a fake  http://purl.obolibrary.org/obo/mondo#should_conform_to

The PROV way of doing it would be very verbose, as there is [no shortcut that adequately describes this relation](https://www.w3.org/TR/prov-o/#description-qualified-terms). the next best thing is `prov:wasDerivedFrom`, but this is more to describe the relation of one dataset that was transferred into another.


Maybe

- dc:conformsTo
    - OMO:generatedFrom  

Is the right way to go. If we agree, I will add these.

**Comment by wdduncan** [comment]
@matentzn If you want to mimic PROV, you could name it `wasGeneratedFrom`. Just a suggestion.

**Comment by matentzn** [comment]
Mimic in what sense? afaics there is no such thing as `wasGeneratedFrom` in PROV!

**Comment by wdduncan** [comment]
I know there is no `wasGeneratedFrom` in PROV. But there is "wasGeneratedBy". By mimic I meant using the "wasGenerated" part.
